modules/gcs/main.tf

```diff
@@ -72,6 +73,14 @@ resource "google_storage_bucket" "buckets" {
       is_locked = lookup(local.retention_policy[each.key], "is_locked", false)
     }
   }
+
+  dynamic logging {
+    for_each = local.logging[each.key] == null ? [] : [""]
+    content {
+      log_bucket = local.logging[each.key]["log_bucket"]
+      log_object_prefix = lookup(local.logging[each.key], "log_object_prefix", "${local.prefix}${lower(each.key)}" )
```

Comment:
Not sure about this : maybe it is better to not set log_object_prefix (and then it will be computed to the bucket name), rather than putting the bucket name if log_object_prefix is not set. But I don't know what is the syntax to do this.

Reply:
I think this should either be a string if you need a specific prefix, or `null` to fallback to the default behaviour. So if you enforce the variable type (comment below), you just need to pass the value specified by the user, which can be `null`:

```hcl
log_object_prefix = local.logging[each.key]["log_object_prefix"]
```